obspy/signal/regression.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------
+# Filename: regression.py
+#  Purpose: Python Module for (Weighted) Linear Regression
+#   Author: Thomas Lecocq
+#    Email: Thomas.Lecocq@seismology.be
+#
+# Copyright (C) 2017 Thomas Lecocq
+# --------------------------------------------------------------------
+"""
+Python Module for (Weighted) Linear Regression.
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (https://www.gnu.org/copyleft/lesser.html)
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+import scipy.optimize
+import numpy as np
+
+
+def linear_regression(xdata, ydata, weights=None, p0=None,
+                      intercept_origin=True):
+    """ Use linear least squares to fit a function, f, to data.
```

Comment:
-> `Use least squares to fit a linear function to data.`  (letter `f` as function name is not used otherwise)